tools/sof_perf_analyzer.py

```diff
@@ -204,8 +212,31 @@ def process_kmsg_file():
                 next_line = next(f)
                 widget_id = next_line.split('|')[0].split(':')[-1].strip()
                 # convert to the same ID format in mtrace
-                widget_id = int('0x' + widget_id[-6:], 16)
+                try:
+                    widget_id = int('0x' + widget_id[-6:], 16)
+                except ValueError:
+                    print(f'no valid widget ID found for {widget_name}, try with bind message')
+                    not_found[widget_name] = ppln_id
+                    continue
```

Comment:
Don't mix `next()` and `continue` in the same `for` loop, this is incredibly hard to follow.

Reply:
BTW there `next_line_if_there_is_one = next(f, None)` if needed. It should probably be used here like this:

```python
next_line = next(f, None)
if not next_line:
   # use some "NotFound" placeholder
   # so the code does not crash later
   # other cleanup if needed.
   raise StopIteration

```

Reply:
1. continue is correct, still need check the following lines in for loop.
2. ok

Reply:
> 1. continue is correct, still need check the following lines in for loop.

I didn't say it was incorrect. I said it was unreadable.

Now I understand the `bind` fallback is actually in a totally different part of the logs. In that case use two different `for` loops. There's no need to look for both at the same time, it makes the code incredibly hard to follow.

Reply:
@btian1 Move `comp_name[str(Component(ppln_id, widget_id))] = widget_name` to the try block, and remove `continue`